webdriver/newsession/basic.py

```diff
@@ -0,0 +1,56 @@
+# META: timeout=long
+
+import uuid
+
+import pytest
+
+from webdriver import error
+
+
+def test_basic(new_session):
+    resp, _ = new_session({"capabilities": {}})
+    assert set(resp.keys()) == {"sessionId", "capabilities"}
+
+
+def test_repeat_new_session(new_session):
+    resp, _ = new_session({"capabilities": {}})
+    with pytest.raises(error.SessionNotCreatedException):
+        new_session({"capabilities": {}})
+
+
+def test_no_capabilites(new_session):
+    with pytest.raises(error.InvalidArgumentException):
+        new_session({})
+
+
+def test_missing_first_match(new_session):
+    resp, _ = new_session({"capabilities": {"alwaysMatch": {}}})
+
+
+def test_missing_always_match(new_session):
+    resp, _ = new_session({"capabilities": {"firstMatch": [{}]}})
+
+
+def test_desired(new_session):
+    with pytest.raises(error.InvalidArgumentException):
+        resp, _ = new_session({"desiredCapbilities": {}})
+
+
+def test_required(new_session):
+    with pytest.raises(error.InvalidArgumentException):
+        resp, _ = new_session({"requiredCapbilities": {}})
+
+
+def test_required_desired(new_session):
+    with pytest.raises(error.InvalidArgumentException):
+        resp, _ = new_session({"requiredCapbilities": {},
+                                               "desiredCapbilities": {}})
+
+
+def test_ignore_required(new_session):
+    new_session({"capabilities": {"requiredCapbilities": {"browserName": "invalid"}}})
+
+
+def test_ignore_desired(new_session):
+    resp, _ = new_session({"capabilities": {"desiredCapbilities": {"pageLoadStrategy": "eager"}}})
```

Comment:
They are different tests. Since requiredCapabilities are required, trying to set an invalid browser name would presumably fail the session, so creating the session implies that we ignored the browser name. For desired capabilities the session will be created anyway, so we need to check that the value was ignored some other way.